library/ssl_tls13_client.c

```diff
@@ -1217,6 +1239,74 @@ int mbedtls_ssl_tls13_write_client_hello_exts( mbedtls_ssl_context *ssl,
     return( 0 );
 }
 
+int mbedtls_ssl_tls13_finalize_write_client_hello( mbedtls_ssl_context *ssl )
+{
+    ((void) ssl);
+#if defined(MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE)
+    mbedtls_ssl_handshake_set_state(
+        ssl, MBEDTLS_SSL_CLIENT_CCS_AFTER_CLIENT_HELLO );
```

Comment:
RFC 8446  section D.4 said: 
-  **If not offering early data**, the client sends a dummy
      change_cipher_spec record (see the third paragraph of [Section 5](https://www.rfc-editor.org/rfc/rfc8446#section-5))
      immediately before its second flight.  This may either be before
      its second ClientHello or before its encrypted handshake flight.
--   **If offering early data**, the record is placed immediately after the
      first ClientHello.
Does that mean we should always send out the ccs, and the difference only exists in :where to send.